SteamKit2/SteamKit2/Steam/Handlers/SteamMatchmaking/SteamMatchmaking.cs

```diff
@@ -0,0 +1,392 @@
+        public bool GetLobbyList( uint appId, List<Lobby.Filter> filters = null, int maxLobbies = -1 )
+        {
+            if ( Client.CellID == null )
+            {
+                return false;
+            }
+
+            var getLobbies = new ClientMsgProtobuf<CMsgClientMMSGetLobbyList>( EMsg.ClientMMSGetLobbyList )
+            {
+                Body =
+                {
+                    app_id = appId,
+                    cell_id = Client.CellID.Value,
+                    public_ip = NetHelpers.GetIPAddress( Client.PublicIP ),
+                    num_lobbies_requested = maxLobbies
+                }
+            };
+
+            if ( filters != null )
+            {
+                foreach ( var filter in filters )
+                {
+                    getLobbies.Body.filters.Add( filter.Serialize() );
+                }
+            }
+
+            Send( getLobbies, appId );
+
+            return true;
+        }
+
+        /// <summary>
+        /// Obtains a <see cref="Lobby"/> by its SteamID, if we already have the data for the lobby locally.
+        /// This method does not send a network request.
+        /// </summary>
+        /// <param name="appId">The ID of app which we're attempting to obtain a lobby for.</param>
+        /// <param name="lobbyId">The SteamID of the lobby that we wish to obtain.</param>
+        /// <returns>The <see cref="Lobby"/> corresponding with the specified app and lobby ID, if we have its data locally. Otherwise, null.</returns>
+        public Lobby GetLobby( uint appId, SteamID lobbyId )
+        {
+            Lobby lobby;
+            var appLobbies = lobbies.GetOrAdd( appId, k => new ConcurrentDictionary<SteamID, Lobby>() );
+            appLobbies.TryGetValue( lobbyId, out lobby );
+            return lobby;
+        }
+
+        /// <summary>
+        /// Sends a matchmaking message for a specific app.
+        /// </summary>
+        /// <param name="msg">The matchmaking message to send.</param>
+        /// <param name="appId">The ID of the app this message pertains to.</param>
+        public void Send( ClientMsgProtobuf msg, uint appId )
+        {
+            if ( msg == null )
+            {
+                throw new ArgumentNullException( nameof(msg) );
+            }
+
+            msg.ProtoHeader.routing_appid = appId;
+            Client.Send( msg );
+        }
+
+        /// <summary>
+        /// Handles a client message. This should not be called directly.
+        /// </summary>
+        /// <param name="packetMsg">The packet message that contains the data.</param>
+        public override void HandleMsg( IPacketMsg packetMsg )
+        {
+            if ( packetMsg == null )
+            {
+                throw new ArgumentNullException( nameof(packetMsg) );
+            }
+
+            Action<IPacketMsg> handler;
+
+            if ( dispatchMap.TryGetValue( packetMsg.MsgType, out handler ) )
```

Comment:
`out var handler`

Reply:
I really dislike that syntax, a surprising amount actually. I typically couldn't care less about syntax (brace position, tabs/spaces, whitespace, semicolons, trailing commas, whatever).

However, I can't stand that syntax. It's not a matter of aesthetics either, it just fundamentally violates scope semantics; a dodgy language hack for not having advanced (multi-line) static type inference.

I'd be fine with having the var declared in the correct scope and it inferring the type when used, but it's not supported.

Granted, it's not my decision to make. I'll make the change as requested. However, that's my reasoning.